### DISCUSSION TITLE
Incorrect behavior of window.print in safari with Next applications

### DISCUSSION THREAD

**Discussion by iamafansev** [vercel/next.js#15254] [Help]
# Bug report

## Describe the bug

When the window.print() method is called in safari, the print window does not open when the button is clicked, but only when the route from the interface is further changed in the application. But everything works correctly in Chrome.

## To Reproduce

[Repository with minimal app from tutorial](https://github.com/iamafansev/blog-with-next)

1. Run app from repository
2. Go to '/'
2. Click on 'print' button from the interface
3. Make sure the print window is not open
4. Click on one of the links from the Blog section to change your route
5. Make sure the print window is open now

## Expected behavior

The print window is expected to open instantly when you press the print button from the interface. But in fact, the window opens only when you further change the route from the interface.

Again, all these steps work correctly in Chrome, but do not work in Safari. I also want to note that everything works correctly in any other react applications, be it server or client side rendering

## System information

- OS: macOS
- Browser: safari
- Version of Next.js: [9.4.4, 9.3.5, 9.2.1]
- Version of Node.js: 14.1.1

**Comment by timneutkens** [comment]
This does not sound related to Next.js and more so a behavior of Safari 🤔

**Reply by iamafansev** [reply]
I assure you that if you follow the specified scenario, you will reproduce this incorrect work. it
also repeated in our enterprise application using next.js and any other application using next.js.
This is not reproducible in all other React apps.

**Reply by iamafansev** [reply]
You should also be alerted that the print window will open only at the time of the route change called from the interface

**Reply by endymion1818** [reply]
I'm trying to do similar but can't get it to work, I've tried the following:

```
<button onClick={() => window.print}>Print</button>
```
And also 
```
{typeof window !== 'undefined' && (
  <button onClick={() => window.print}>Print</button>
)}
```
But neither work either in dev or build mode.
